app/parse.py

```diff
@@ -15,8 +40,106 @@ class Product:
     num_of_reviews: int
 
 
+def get_product(product_card: WebElement) -> Product:
+    return Product(
+        title=product_card.find_element(By.CLASS_NAME, "title").get_attribute("title"),
+        description=product_card.find_element(By.CLASS_NAME, "description").text,
+        price=float(
+            product_card.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+        ),
+        rating=len(product_card.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        num_of_reviews=int(
+            product_card.find_element(By.CLASS_NAME, "review-count").text.split()[0]
+        ),
+    )
+
+
+def get_all_from_dynamic_page(url: str) -> list[Product]:
+    logging.info(f"Get all products dynamic page'{url}'")
+    driver = get_driver()
+    driver.get(urljoin(BASE_URL, url))
+    try:
+        close_button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+        driver.execute_script("arguments[0].scrollIntoView(true);", close_button)
+        close_button.click()
+        logging.info("Closed cookie banner")
+    except NoSuchElementException:
+        logging.info("Cookie banner not found")
+    except ElementNotInteractableException:
+        logging.info("The cookie banner was unclickable")
+    try:
+        more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        while True:
+            try:
+                if not more.is_displayed():
+                    break
+                ActionChains(driver).move_to_element(more).click(more).perform()
+                logging.info("Clicked 'Load More' button")
+                time.sleep(1)
+                more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+            except ElementClickInterceptedException:
+                logging.warning(
+                    "Load More button was unclickable;"
+                    " retrying after closing any obstructive elements"
+                )
+                cookie_button = driver.find_element(By.CLASS_NAME, "acceptCookies")
+                if cookie_button:
+                    cookie_button.click()
```

Comment:
The code attempts to click the cookie button again if the 'Load More' button is unclickable. Ensure that this logic doesn't lead to an infinite loop if the cookie button is not the cause of the issue. Consider adding a break condition or a maximum retry limit.